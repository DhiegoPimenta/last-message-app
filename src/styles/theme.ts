export default {
    grid: {
        container: '130rem',
        gutter: '3.2rem',
    },
    border: {
        radius: '1rem',
    },
    font: {
        family: `Poppins, Ubuntu, -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Oxygen, Cantarell, 'Open Sans', 'Helvetica Neue', sans-serif`,
        light: 300,
        normal: 400,
        bold: 600,
        extrabold: 800,
        sizes: {
            xsmall: '1.4rem',
            small: '1.8rem',
            medium: '2.4rem',
            large: '3.2rem',
        },
    },
    colors: {
        primary: '#64CBC5',
        secondary: '#0030FF',
        terciary: '#A0DFDC',
        white: '#FFFFFF',
        gray: '#CECECE',
        danger: '#FBA9B0',
        black: '#444444',
        overlay: 'rgba(0, 0, 0, 0.3)',
    },
    spacings: {
        xxsmall: '0.8rem',
        xsmall: '1.6rem',
        small: '2.4rem',
        medium: '3.2rem',
        large: '4.0rem',
        xlarge: '4.8rem',
        xxlarge: '5.6rem',
    },
    layers: {
        base: 10,
        menu: 20,
        overlay: 30,
        modal: 40,
        alwaysOnTop: 50,
    },
    transition: {
        default: '0.3s ease-in-out',
        fast: '0.1s ease-in-out',
    },
} as const
