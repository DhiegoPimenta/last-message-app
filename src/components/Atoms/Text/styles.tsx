import styled, { css } from 'styled-components'
import theme from 'styles/theme'
import { TextProps } from '.'

const wrapperModifiers = {
    fullWidth: () => css`
        text-align: center;
    `,
}

export const Text = styled.text<
    Pick<TextProps, 'color' | 'fullWidth' | 'size'>
>`
    ${({ color, fullWidth, size }) => css`
        color: ${theme.colors[color!]};
        font-size: ${theme.font.sizes[size!]};
        width: 100%;
        display: block;
        font-weight: ${theme.font.normal};
        ${!!fullWidth && wrapperModifiers.fullWidth()}
    `}
`
