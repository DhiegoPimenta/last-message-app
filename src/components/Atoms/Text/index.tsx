import { ThemeColors, ThemeFontSizes } from 'styles/global'
import * as S from './styles'

export type TextProps = {
    children: string
    color?: ThemeColors
    fullWidth?: boolean
    tagName?: any
    size?: ThemeFontSizes
}

const Text = ({
    children,
    color = 'primary',
    fullWidth,
    tagName = 'p',
    size = 'medium',
}: TextProps) => {
    return (
        <S.Text
            color={color}
            fullWidth={fullWidth}
            size={size}
            as={tagName}
            dangerouslySetInnerHTML={{ __html: children }}
        />
    )
}

export default Text
