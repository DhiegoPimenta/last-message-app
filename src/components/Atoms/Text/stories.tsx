import { Story, Meta } from '@storybook/react/types-6-0'

import Text, { TextProps } from '.'

export default {
    title: 'Atoms/Text',
    component: Text,
} as Meta

export const Basic: Story<TextProps> = (args) => <Text {...args} />
Basic.args = {
    children: 'texto',
}
