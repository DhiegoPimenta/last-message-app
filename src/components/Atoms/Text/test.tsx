import { screen } from '@testing-library/react'
import theme from 'styles/theme'
import { renderWithTheme } from 'utils/tests/helpers'

import Text from '.'

describe('<Text />', () => {
    it('should render text', () => {
        const { container } = renderWithTheme(<Text tagName="h2">text</Text>)

        expect(screen.getByRole('heading')).toBeInTheDocument()

        expect(container.firstChild).toMatchSnapshot()
    })

    it('should render mediun and primary color as default', () => {
        renderWithTheme(<Text tagName="h2">Text</Text>)

        expect(screen.getByText(/text/i)).toBeInTheDocument()

        expect(screen.getByText(/text/i)).toHaveStyle({
            'font-size': '2.4rem',
            color: theme.colors.primary,
        })
    })

    it('should render change size', () => {
        renderWithTheme(<Text size="large">text</Text>)

        expect(screen.getByText(/text/i)).toBeInTheDocument()
        expect(screen.getByText(/text/i)).toHaveStyle({
            'font-size': '3.2rem',
        })
    })

    it('should render change color', () => {
        renderWithTheme(<Text color="danger">text</Text>)

        expect(screen.getByText(/text/i)).toBeInTheDocument()
        expect(screen.getByText(/text/i)).toHaveStyle({
            color: theme.colors.danger,
        })
    })

    it('should render full width', () => {
        renderWithTheme(<Text fullWidth>text</Text>)

        expect(screen.getByText(/text/i)).toBeInTheDocument()
        expect(screen.getByText(/text/i)).toHaveStyle({
            width: '100%',
            'text-align': 'center',
        })
    })
})
