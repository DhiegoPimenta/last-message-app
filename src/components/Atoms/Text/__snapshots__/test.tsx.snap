// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Text /> should render text 1`] = `
.c0 {
  color: #64CBC5;
  font-size: 2.4rem;
  width: 100%;
  display: block;
  font-weight: 400;
}

<h2
  class="c0"
  color="primary"
>
  text
</h2>
`;
